### PR TITLE
Register page

### PR DESCRIPTION
**Pull request by pkakelas**
screenshot from 2014-08-12 21 00 01
screenshot from 2014-08-12 21 02 13

### CODE REVIEW

File: helpers/html.php [hunk 1]
```diff
@@ -137,10 +137,17 @@ public static function getRESTMethodIdempotence( $method ) {
...
                 echo htmlspecialchars( $for );
-            ?>"><?php
+                ?>" <?php
+                foreach ( $attributes as $key => $value ) {
+                    echo $key;
+                    ?>="<?php
+                        echo htmlspecialchars( $value );
+                    ?>" <?php
+                }
+                ?>><?php
```
Comment (dionyziz) [inline helpers/html.php:150]: DRY: Split attribute generation into helper function.

**Comment by dionyziz** [comment]
Screenshot.

File: views/user/create.php [hunk 2]
```diff
@@ -1,82 +1,117 @@
...
+        <div class="form-group"><?php
+            if ( isset( $email_empty ) ) {
+                $self->createError( 'Please type an email.' );
+            }
+            if ( isset( $email_invalid ) ) {
+                $self->createError( 'Email is not valid' );
+            }
+            if ( isset( $email_used ) ) {
+                $self->createError( 'Email is already in use' );
+            }
```
Comment (dionyziz) [inline views/user/create.php:74]: DRY: Error handling pattern can be abstracted, should we make an issue?

File: views/user/create.php [hunk 3]
```diff
@@ -1,82 +1,117 @@
...
+            if ( isset( $username_invalid ) ) {
+                $self->createError( 'Usernames can only have numbers, letters, "." and "_"' );
+            }
+            if ( isset( $username_used ) ) {
+                $self->createError( 'Username already exists' );
+                $username_value = "";
+            }
+            else if ( isset( $_SESSION[ 'create_post' ][ 'username' ] ) ) {
+                $username_value = $_SESSION[ 'create_post' ][ 'username' ];
+                unset( $_SESSION[ 'create_post' ][ 'username' ] );
```
Comment (dionyziz) [inline views/user/create.php:30]: DRY: Refilling form details can be abstracted, can we make an issue?

File: views/user/create.php [hunk 4]
```diff
@@ -1,82 +1,117 @@
...
+                    'placeholder' => 'Email'
+                ] );
+            ?></div>
+        </div>
+        <div class="form-group form-inline"><?php
+            $self->createLabel( 'day', 'Birthday', [ 'class' => 'col-sm-2 control-label' ] );
+            ?><div id='dob' class="row"><?php
+                $days = createSelectPrepare( range( 1, 31 ), 'day' );
+                $self->createSelect( $days, 'day', '', '', [ 'class' => 'form-control dob-input' ] );
+                $months = createSelectPrepare( range( 1, 12 ), 'month' );
```
Comment (dionyziz) [inline views/user/create.php:88]: Can we see a screenshot of how this looks with month names spelled out in full?

File: views/user/create.php [hunk 5]
```diff
@@ -1,82 +1,117 @@
...
-        $years = createSelectPrepare (
-            range( $current_year - $config[ 'age' ][ 'min' ], $current_year - $config[ 'age' ][ 'max' ] ),
-            'Select Year'
-        );
-        $self->createSelect( $years, 'year' );
-        $countriesSelectArray[] = 'Select Country';
-        foreach ( $countries as $key => $country ) {
-            $countriesSelectArray[ $country->shortname ] = $country->name;
-        }
-        $self->createSelect( $countriesSelectArray, 'countryShortname', $location );
```
Comment (dionyziz) [inline views/user/create.php:77]: `$location` is not used in the new patch.

File: views/user/create.php (hunk 3 above)
Comment (pkakelas) [inline views/user/create.php:30]: Done

File: helpers/html.php (hunk 1 above)
Comment (pkakelas) [inline helpers/html.php:150]: Ok, I'll fix them when this pr and the update page pr get merged.

File: views/user/create.php (hunk 2 above)
Comment (pkakelas) [inline views/user/create.php:74]: Done

**Pull request merged**